tls/s2n_server_key_exchange.c

```diff
@@ -161,16 +159,46 @@ int s2n_kem_server_key_recv_read_data(struct s2n_connection *conn, struct s2n_bl
     kem_data->raw_public_key.size = key_length;
 
     data_to_verify->size = sizeof(kem_extension_size) + sizeof(kem_public_key_size) + key_length;
-
     return 0;
 }
 
-int s2n_kem_server_key_recv_parse_data(struct s2n_connection *conn, union s2n_kex_raw_server_data *raw_server_data)
+int s2n_kem_server_key_recv_parse_data(struct s2n_connection *conn, struct s2n_kex_raw_server_data *raw_server_data)
 {
     s2n_dup(&raw_server_data->kem_data.raw_public_key, &conn->secure.s2n_kem_keys.public_key);
     return 0;
 }
 
+int s2n_hybrid_server_key_recv_read_data(struct s2n_connection *conn, struct s2n_blob *total_data_to_verify, struct s2n_kex_raw_server_data *raw_server_data)
+{
+    const struct s2n_kex *kex = conn->secure.cipher_suite->key_exchange_alg;
+    const struct s2n_kex *hybrid_kex_0 = kex->hybrid[0];
+    const struct s2n_kex *hybrid_kex_1 = kex->hybrid[1];
+
+    /* Keep a copy to the start of the whole structure for the signature check */
+    total_data_to_verify->data = s2n_stuffer_raw_read(&conn->handshake.io, 0);
+    notnull_check(total_data_to_verify->data);
+
+    struct s2n_blob data_to_verify_0 = {0};
+    GUARD(s2n_kex_server_key_recv_read_data(hybrid_kex_0, conn, &data_to_verify_0, raw_server_data));
+
+    struct s2n_blob data_to_verify_1 = {0};
+    GUARD(s2n_kex_server_key_recv_read_data(hybrid_kex_1, conn, &data_to_verify_1, raw_server_data));
+
+    total_data_to_verify->size = data_to_verify_0.size + data_to_verify_1.size;
+    return 0;
+}
+
+int s2n_hybrid_server_key_recv_parse_data(struct s2n_connection *conn, struct s2n_kex_raw_server_data *raw_server_data)
+{
+    const struct s2n_kex *kex = conn->secure.cipher_suite->key_exchange_alg;
```

Comment:
Same comment, can we add `notnull_check` for `conn` and `conn->secure.cipher_suite` here?

Reply:
Added to the three new hybrid methods. What is the guide for using these safety checks? Should I add similar checks to all the other client and server key exchange methods?